lib/faraday/connection.rb

```diff
@@ -522,7 +522,9 @@ def build_exclusive_url(url = nil, params = nil, params_encoder = nil)
         base = base.dup
         base.path = "#{base.path}/" # ensure trailing slash
       end
+      url = url && URI.parse(url.to_s).opaque ? url.to_s.gsub(':', '%3A') : url
       uri = url ? base + url : base
+      uri = URI.parse(CGI.unescape(uri.to_s))
```

Comment:
Not sure whether final uri should be unescaped

Reply:
That's a good point, this may cause issues in case the URL contains escaped characters like spaces? Escaped columns may also be part of the URL, so I'm not sure unescaping here is a good idea at all.
E.g. `Faraday.get('https://service.com/path%20with%20spaces')